### ISSUE TITLE
partial pipeline retained after complete?

### ISSUE THREAD

**Issue by tisdall** [python-social-auth/social-core#112]
I upgraded from an older version of PSA.  The older one just retained data in the Django session and the new one uses a separate table indexed by a token.  I've managed to figure out how to fetch this partial pipeline in a regular view, but that partial pipeline seems to be retained after the person has gone through the "/complete/[backend]" step successfully.  Is this a bug?  If I recall correctly, the old one would remove the session information after the pipeline was successfully completed.

What I'm trying to do is detect whether I'm in the middle of a pipeline for a user and redirect them to "/complete/[backend]/" but since the pipeline is never cleared away it's always redirecting.  Is there some other way to detect if someone is part way through a pipeline?

**Comment by omab** [comment]
Fixed by https://github.com/python-social-auth/social-core/commit/def3aa8bb3b048017069b9c7df6a07d8c78df7f4

**Comment by newenegue** [comment]
@omab When will this fix be release?

**Comment by tisdall** [comment]
Here's what I've been using to get around this issue until a new release has been made:

```python
        strategy = load_strategy(request)
        partial = strategy.partial_load(partial_pipeline_token)

        #  temp fix until next python-social-auth released (one after 1.7.0)
        if partial is None:
            # should only be None if no longer in social-auth table so clear session token
            strategy.clean_partial_pipeline(partial_pipeline_token)
```

**Comment by newenegue** [comment]
I ended bypassing the session by using `@partial_step` (https://github.com/python-social-auth/social-core/issues/145) and passing the `partial_token` in the url params.
